exercises/forth/example.go

```diff
@@ -0,0 +1,243 @@
+// Package forth implements a tiny subset of the Forth language.
+package forth
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+type operatorFn func(stack *[]int) error
+
+type operatorId byte
+
+const (
+	addOp operatorId = iota
+	subOp
+	mulOp
+	divOp
+	dropOp
+	dupOp
+	swapOp
+	overOp
+	constOp
+	userDefOp
+	endDefOp
+)
+
+type operatorTyp struct {
+	fn operatorFn
+	id operatorId
+}
+
+func Forth(input []string) (result []int, err error) {
+	if len(input) == 0 {
+		return result, nil
+	}
+
+	stack := make([]int, 0, 8)
+	userDefs := make(map[string][]operatorTyp, 8)
+	for _, phrase := range input {
+		// Parse one phrase of input, building up an operator list.
+		opList, err := parse(phrase, userDefs)
+		if err != nil {
+			return nil, err
+		}
+		// Perform any operators from that phrase, updating stack.
+		for _, opr := range opList {
+			err = opr.fn(&stack)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return stack[:len(stack)], nil
+}
+
+// parse given phrase, returning an operator list, and updating
+// a userDefs map for any user definifion of words in phrase.
+func parse(phrase string, userDefs map[string][]operatorTyp) (oplist []operatorTyp, err error) {
+	words := strings.FieldsFunc(phrase,
+		func(r rune) bool {
+			return unicode.IsSpace(r) || unicode.IsControl(r)
+		})
+
+	var t int // t is token index into words[]
```

Comment:
hmm, any particular reason why this `var t int` line needs to be here? What if it were removed, leaving just the comment `t is token index...` ? would anything bad happen? I think `t` isn't used outside the for loop, but maybe there is something subtle.